### ISSUE TITLE
Add distance dependent damage to d2k

### ISSUE THREAD

**Issue by CH4Code** [OpenRA/OpenRA#14692]
Like MustaphaTR noted on #14681 units in d2k do less damage if they are further away from their targets.
I got curious and did some testing on d2k (grundtmod edition) with a combat tank:
EDIT: the coloring is wrong (blue values = red graph)
![distancedamage](https://user-images.githubusercontent.com/16046997/34675876-33a19c3e-f48b-11e7-8c99-0deb1b172aef.png)

seems to be almost linear with dmg(dist) = (full dmg) * (100% + loss * dist)  where loss= -10% for the harkonnen combat tank.

...unfortunately the slope is different for different units; siege tank for example has a low loss= -3% and trike a very steep one with loss= -17%. 
The loss also does not depend on max range as the missile tank (which has more range than a siege tank) has a geater loss= -8%...
Seems just like an extra table with arbitrarily picked values (like "armor" or "damage type"). Maybe we can get that with TibEd?

Note that the yellow value is extrapolated and has not been measured.

**Comment by pchote** [comment]
> The loss also does not depend on max range

It would be good to confirm this by repeating the experiment after using tibed to edit the max range of the same unit.

**Comment by CH4Code** [comment]
> You should also test the same unit vs different structures and units to make sure that it doesn't depend on the target.
Seems to be all over the place. trike loss becomes almost exponential when facing a quad (light armor).
On the other hand tank damage is pretty linear vs quad.
Doesn't seem this can be handled easily. Maybe they even had different functions for different weapons :S

**Comment by pchote** [comment]
There may be an inaccuracy factor hidden in all this: at larger distances a projectile may be less likely to hit directly which will then affect the damage dealt depending on how the original game handled hit shapes.

**Comment by GraionDilach** [comment]
I'd like to mention here that TS also had used a logic akin - https://www.modenc.renegadeprojects.com/Degenerates - which IIRC was used on the Cyborg Commando.

**Comment by NimoStar** [comment]
This is a property of the warhead AFAIK. Tiberian Sun has also a "dispersion" mechanism for this - the farther away the unit fires, the damage is more spread thus you believe it is "losing" damage with distance. That's why it's high for some units and low for others.

**Comment by pchote** [comment]
Fixed by #18009 / #18010 / #18023.
